File: casser_code_pin.py
import random
import time

# 1. Code secret aléatoire (ou décommenter pour saisie manuelle)
code_secret = random.randint(0, 9999)
# code_secret = int(input("Définissez le code PIN à 4 chiffres à trouver : "))

print("Début du programme")
start_time = time.time()

trouve = False
i = 0
limite_tentatives = 10000

# 2. Boucle de force brute
while i < 10000 and not trouve and i < limite_tentatives:
    print(i, end=" ")

    # Temporisation (Mission 5) : 5 ms = 0.005 s / 1 s = 1.0 s
    # time.sleep(0.005)

    if i == code_secret:
        trouve = True
        print(f"\n\nLe code secret est : {i}")

    i += 1

if not trouve:
    print("\n\nÉchec : Nombre maximal de tentatives atteint !")

print("Fin du programme")

# 3. Calcul de la durée
duree = time.time() - start_time
print(f"durée du programme: {duree:.6f} secondes")

File: casser_code_pin.php
<?php

// 1. Code secret aléatoire (ou décommenter pour saisie manuelle)
$codeSecret = rand(0, 9999);
// echo "Définissez le code PIN à 4 chiffres à trouver : ";
// $codeSecret = (int)trim(fgets(STDIN));

echo "Début du programme\n";
$start = microtime(true);

$trouve = false;
$i = 0;
$limiteTentatives = 10000;

// 2. Boucle de force brute
while ($i < 10000 && !$trouve && $i < $limiteTentatives) {
    // Affichage formaté sur 4 chiffres
    echo sprintf('%04d', $i) . " ";

    // Temporisation (Mission 5) : usleep prend des microsecondes (5000 µs = 5 ms)
    // usleep(5000);

    if ($i === $codeSecret) {
        $trouve = true;
        echo "\n\nLe code secret est : " . sprintf('%04d', $i) . "\n";
    } else {
        $i++;
    }
}

if (!$trouve) {
    echo "\n\nÉchec : Nombre maximal de tentatives atteint !\n";
}

echo "Fin du programme\n";

// 3. Calcul de la durée
$duree = microtime(true) - $start;
echo "durée du programme: " . number_format($duree, 6) . " secondes\n";
echo "Fin du programme\n";

// 3. Calcul de la durée
$duree = microtime(true) - $start;
echo "durée du programme: " . number_format($duree, 6) . " secondes\n";
